app/controllers/UnderpaymentSummaryController.scala

```diff
@@ -0,0 +1,97 @@
+import views.ViewUtils.displayMoney
+import views.html.UnderpaymentSummaryView
+
+import scala.concurrent.Future
+
+@Singleton
+class UnderpaymentSummaryController @Inject()(identify: IdentifierAction,
+                                              getData: DataRetrievalAction,
+                                              requireData: DataRequiredAction,
+                                              mcc: MessagesControllerComponents,
+                                              view: UnderpaymentSummaryView)
+  extends FrontendController(mcc) with I18nSupport {
+
+  val onLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+
+    val customsDuty = request.userAnswers.get(CustomsDutyPage).map(
+      summaryList(_, Messages("underpaymentSummary.customsDuty.title"), controllers.routes.CustomsDutyController.onLoad))
+
+    val importVat = request.userAnswers.get(ImportVATPage).map(
+      summaryList(_, Messages("underpaymentSummary.importVat.title"), controllers.routes.ImportVATController.onLoad))
+
+    val exciseDuty = request.userAnswers.get(ExciseDutyPage).map(
+      summaryList(_, Messages("underpaymentSummary.exciseDuty.title"), controllers.routes.ExciseDutyController.onLoad))
+
+    Future.successful(Ok(view(customsDuty, importVat, exciseDuty, controllers.routes.UnderpaymentTypeController.onLoad)))
+  }
+
+  private[controllers] def summaryList(underpayment: UnderpaymentAmount, key: String, changeAction: Call)(implicit messages: Messages): SummaryList = {
+    SummaryList(
+      classes = "govuk-!-margin-bottom-9",
+      rows = Seq(
+        SummaryListRow(
+          key = Key(
+            content = Text(messages("underpaymentSummary.originalAmount")),
+            classes = "govuk-!-width-two-thirds govuk-!-padding-bottom-0"
+          ),
+          value = Value(
+            content = HtmlContent(displayMoney(underpayment.original)),
+            classes = "govuk-!-padding-bottom-0"
+          ),
+          actions = Some(Actions(
+            items = Seq(
+              ActionItem(changeAction.url, Text(messages("underpaymentSummary.change")), Some(key))
+            ),
+            classes = "govuk-!-padding-bottom-0")
+          ),
+          classes = "govuk-summary-list__row--no-border"
+        ),
+        SummaryListRow(
+          key = Key(
+            content = Text(messages("underpaymentSummary.amendedAmount")),
+            classes = "govuk-!-width-two-thirds govuk-!-padding-top-0"
+          ),
+          value = Value(
+            content = HtmlContent(displayMoney(underpayment.amended)),
+            classes = "govuk-!-padding-top-0"
+          )
+        ),
+        SummaryListRow(
+          key = Key(
+            content = Text(messages("underpaymentSummary.dueToHmrc", key)),
+            classes = "govuk-!-width-two-thirds"
+          ),
+          value = Value(content = HtmlContent(displayMoney(underpayment.amended - underpayment.original))),
```

Comment:
comma at the end of line is what's breaking the PR builder lol

Reply:
I'm surprised that compiles but it isn't necessary so I have removed it.